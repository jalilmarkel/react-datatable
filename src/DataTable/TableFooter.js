import styled from 'styled-components';

const TableFooter = styled.nav`
  display: flex;
  flex: 1 1 auto;
  justify-content: flex-end;
  align-items: center;
  min-height: 56px;
  padding-right: 8px;
  padding-left: 8px;
  width: 100%;
  border-top: 1px solid ${props => props.theme.rows.borderColor};
`;
export default TableFooter;
