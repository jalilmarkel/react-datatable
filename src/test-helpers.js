import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

