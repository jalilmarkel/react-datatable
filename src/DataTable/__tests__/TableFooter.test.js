import 'jest-styled-components';
import React from 'react';
import { cleanup } from 'react-testing-library';
import { renderWithTheme } from '../../test-helpers';
import TableFooter from '../TableFooter';

afterEach(cleanup);

test('should render correctly', () => {
  const { container } = renderWithTheme(<TableFooter />);

  expect(container.firstChild).toMatchSnapshot();
});
