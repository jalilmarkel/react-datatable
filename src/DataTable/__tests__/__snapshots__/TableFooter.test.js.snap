// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 56px;
  padding-right: 8px;
  padding-left: 8px;
  width: 100%;
  border-top: 1px solid rgba(0,0,0,.12);
}

<nav
  class="c0"
/>
`;
